### PR TITLE
fix potential reconnect issues + cleanups

### PR DESCRIPTION
**Pull request by kof**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,59 +23,63 @@ function WebSocketWrapper(url) {
...
+	this._socket = this._connect();
 }
 WebSocketWrapper.prototype = Object.create(Emitter.prototype);
 
-// wrap the states
-WebSocketWrapper.prototype.CONNECTING = 0;
-WebSocketWrapper.prototype.OPEN = 1;
-WebSocketWrapper.prototype.CLOSING = 2;
-WebSocketWrapper.prototype.CLOSED = 3;
+WebSocketWrapper.CONNECTING = 0;
```
Comment (kof) [inline index.js:30]: This constants should not be referenced on the instance.

File: index.js [hunk 2]
```diff
@@ -23,59 +23,63 @@ function WebSocketWrapper(url) {
...
 WebSocketWrapper.prototype.send = function WebSocketWrapper_send(msg) {
-	if (this.readyState === this.OPEN)
-		this._send(msg);
+	if (this.readyState === WebSocketWrapper.OPEN) {
+		// Calling send() on a closed WebSocket object which reports an open
+		// readyState causes a crash. This scenario can occur when returning to
+		// a backgrounded page which received data and then closed when in the
+		// backgrounded state.
+		// https://gist.github.com/mloughran/2052006
+		setTimeout(function () {
```
Comment (kof) [inline index.js:51]: this timeout should be the main fix

**Comment by feychou** [comment]
so what seemed to be the problem here?

**Comment by kof** [comment]
Please read my comments in code and the link on line 50 :)

**Comment by feychou** [comment]
I mean, in the entire pull request.

File: index.js (hunk 2 above)
Comment (feychou) [inline index.js:51]: is setting a time out the only option to fix this? or just a temporary solution to check if this is the problem?

Comment (kof) [inline index.js:51]: its the only known hack to me.

Comment (feychou) [inline index.js:51]: fair enough ^^

Comment (kof) [inline index.js:51]: Its a hack for sure, but there is no valid fix for this.

Comment (kof) [inline index.js:51]: The issue is just very low level.

**Comment by feychou** [comment]
:+1:

side note: two different commits for fix and cleanup would make things clearer.

**Comment by kof** [comment]
yep

File: index.js (hunk 1 above)
Comment (sk7) [inline index.js:30]: just curious: what happened to the other constants? they are not used anywhere?

Comment (feychou) [inline index.js:30]: They are below, I also had the impression they were gone.

Comment (sk7) [inline index.js:30]: ah ok didn't see that ^^

Comment (kof) [inline index.js:30]: just moved them, and yes, 2 of them are not used.

**Pull request merged**